src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -45,5 +47,1075 @@ class Outer
             comp = CreateCompilation(source);
             comp.VerifyDiagnostics();
         }
+
+        [Fact]
+        public void Nested_02()
+        {
+            var source = """
+                file class Outer
+                {
+                    class C { }
+                }
+                """;
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
+            comp.VerifyDiagnostics(
+                // (1,12): error CS8652: The feature 'file types' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                // file class Outer
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "Outer").WithArguments("file types").WithLocation(1, 12));
+
+            comp = CreateCompilation(source);
+            comp.VerifyDiagnostics();
```

Comment:
nit: It would be good to verify the visibility and effective accessibility of the nested type